### PR TITLE
(doc) update new release doc

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

File: Documentation/design/release.md [hunk 1]
```diff
@@ -1,32 +1,93 @@
...
+This should generate no change(s) in `manifests` folder. If there are no change(s) in `manifests` folder then you can proceed with the new release. 
+You can clean your workspace 
+```bash
+git checkout -- .
+git clean -df
+```
+
+If you see change(s) in `manifests` folder, then:
+* STOP the release process
+* Consult the team and backport the changes in `manifests` folder into the templates.
```
Comment (jpeeler) [inline Documentation/design/release.md]: Whoa - have people been putting changes into the manifests without using the templates? I assumed that when this happened the templates were modified and no make release was done.

Comment (tkashem) [reply Documentation/design/release.md]: Everything went fine for me, I just added it as a precaution. :)

**Comment by tkashem** [comment]
> Thanks for making this clearer!

I must thank @dinhxuanvu, he helped me with the release and gave me the idea to update the readme :)

**Pull request merged**